---
title: "Microbial dIversity and Network Analysis with MINA"
author: "Rui Guan"
date: "`r Sys.Date()`"
abstract: >
    An increasing number of microbiome datasets are generating with the help of
    rapidly developing sequencing technologies. Currently, analysis of taxonomic
    profiling data is mainly conducted using composition-based methods, which
    ignores the interactions between community members and limits the study of
    community dynamics. The goal of MINA is to provide a framework for microbial
    community analysis based on network structures with statistical tests.
    MINA package version: `r packageVersion("mina")`
output:
    rmarkdown::html_vignette:
        highlight: pygments
        toc: true
        fig_width: 5
vignette: >
  %\VignetteIndexEntry{Microbial dIversity and Network Analysis with MINA}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\usepackage[utf8]{inputenc}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
# Overview

The package is used for microbial profiling data analysis, including community
similarity / dissimilarity / distance calculation and network comparison.

# Input data
MINA package expects count data (e.g. OTU table or ASV table) to represent
community profiling results and a descriptive table which indicates the
information of each sample. In the quantitative table, each row contains one
composition in the community and each column represent one profiled sample. In
the descriptive table, same samples as rows in quantitative table should be
included in the column "SampleID".


# About this vignette

## Typesetting Legend <a id="sec:typeset-legend"></a>

- **bold** - Bold is used for emphasis.
- *italics* - Italics are used for package names, and special words, phrases.
- `code font` - The font for code, usually courrier-like, but depends on the
  theme.
- `myFun()` - Code font word with `()` attached at the right-end, is a functioni
  name.
- [Hyperlink](#sec:typeset-legend) - Hyperlinks are
clickable text that will jump to sections and external pages.

## Import data

The first step is to import quantitative table of microbial community profiling
data (e.g. OTU table) and corresponding descriptive file (e.g. design file) into
the object.
To load data, use the `new()` for all avaliable file or `add_tab()` and
`add_des()` to import them seperately.

For the quantitative file (e.g. OTU table), rows are compositions of the
community and columns are samples. The corresponding descriptive file(e.g.
design file) should
have the same samples as quantitative table. The first column of descriptive
file is "**Sample_\_ID**" which contains IDs of samples.

#### Example

Import files together:

```{r, eval=FALSE}
m <- new(mina, tab = "otu_table.txt", des_tab = "design.txt")
```
Import files seperately:
```{r, eval=FALSE}
m1 <- new(mina)
m1 <- addTab(m1, "otu_table.txt")
m1 <- addDes(m1, "design.txt")
```
### Normalize quantitative table


## Supported Analysis


## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
